src/LuaPiGui.cpp

```diff
@@ -65,6 +66,13 @@ void *pi_lua_checklightuserdata(lua_State *l, int index)
 	return nullptr;
 }
 
+void pi_lua_generic_pull(lua_State *l, int index, ImVec2 &vec)
```

Comment:
Mmmmhh... I don't like too much mixing these types...
I mean that when I removed ImVec2 pull/pop I explicitly copy/pasted
the conversion to avoid this kind of things to kick in: it was IMO a
(primitive?) way to confine ImVec2 usage.

Anyway, if that thing is not appreciated and thus you would this function
in master, sure you should use it extensively: thus changing all the involved
lines in order to standardize things and increase visibility, thus avoiding any
subsequent programmer from implementing its owns methods.

Reply:
Had I caught it earlier, I would have suggested that you define a conversion function instead of going to the work of rewriting every function that needs an ImVec2, but it is what it is. I intend to go back through and fix up each function which does the conversion manually, so don't worry; I'll establish a common convention.

Considering that the alternatives are to add a UDC to either ImVec2 or vector2d, or to manually instantiate multiple ImVec2s in every single function that needs to pull a vector from lua, I think this is an acceptable solution.

Reply:
...Well, as you wish... Not a problem but you answered yourself:

`draw_list->AddLine(ImVec2(p1.x, p1.y), ImVec2(p2.x, p2.y), color, thickness);`

(line 329, `LuaPiGui.cpp`)

Why this shouldn't suffice? Compiler will throw away useless bytes of code and you're
away from any mixing of these types or from `ImGui` types proliferation...

Reply:
Because it's a significant amount of extra work and complexity, both for anyone maintaining the existing code, and anyone adding new code to the PiGui implementation. The entire point of a conversion function like `LuaPull` is to hide and consolidate the verbose code that takes one input (e.g. a Lua value) and outputs the same content in a different format (e.g. an ImVec2).

Reply:
Good, so it seems I'm the only one having the will to prevale over that copy-paste-hell. :sweat_smile: 

I'm prone to suggest you the use of a couple of "helper function"; I mean something like
```
vector2 ImVec2vector( ImVec2 )
vector3 ImVec2vector( ImVec3 )
```
and back, with a name that helps separating their purposes from `LuaPush/Pop(****)` methods, because in general

> The entire point of a conversion function like LuaPull 

...is to pull/pop from Lua, and if you hide that complexity (IMHO), you end having no idea of what type of data are you working with (...a side effect of overloading) as demostrated in recent PR where has been was really difficult to explain to all people (including myself) what happened there...

Reply:
And yet, using those functions still adds extra complexity (and work!) to what is solved by the existing adapter function, `LuaPull`. Considering that the specific overload is defined in only one CPP file, it's not like we're hiding where the conversion is happening or what underlying datatype is being pulled in - we're just reducing the amount of work it takes to pull a vector type from Lua and pass it to ImGui.

I'm sure @impaktor and @fluffyfreak will back me up when I say that something that makes the code more maintainable and reduces the amount of unnecessary copy-and-paste boilerplate you need to write for _every single function_ is something much more preferable for Pioneer then requiring our already busy contributors to manually remember and perform the conversion every time.